src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_params.py

```diff
@@ -239,17 +241,21 @@ def register_source_uri_arguments(scope):
 for item in ['blob', 'file', 'queue', 'table']:
     register_cli_argument('storage account show-connection-string', '{}_endpoint'.format(item), help='Custom endpoint for {}s.'.format(item))
 
+register_cli_argument('storage account create', 'account_type', help='The storage account type', **enum_choice_list(get_versioned_models(ResourceType.MGMT_STORAGE_STORAGE_ACCOUNTS, model='AccountType')))
+
 register_cli_argument('storage account create', 'account_name', account_name_type, options_list=('--name', '-n'), completer=None)
 
-register_cli_argument('storage account create', 'kind', help='Indicates the type of storage account.', **enum_choice_list(Kind))
+register_cli_argument('storage account create', 'kind', help='Indicates the type of storage account.', **enum_choice_list(get_versioned_models(ResourceType.MGMT_STORAGE_STORAGE_ACCOUNTS, model='Kind')))
```

Comment:
`**enum_choice_list(get_versioned_models(ResourceType.MGMT_STORAGE_STORAGE_ACCOUNTS, model='Kind'))`

IMO this syntax is not ready for prime time. Compare to:
`**enum_choice_list(Kind)` or `**enum_choice_list(['Blob', 'Normal'])` 
I would not expect any third party contributors to get this right. It's hard and cumbersome to do the right thing in this case, which is not a good thing. We should look at ways to simplify this.

Also, what happens for the API versions that don't even support "Kind"?